app/mailers/application_mailer.rb

```diff
@@ -1,7 +1,9 @@
 require 'uri'
 
 class ApplicationMailer < ActionMailer::Base
+  after_filter :set_configurations unless Setting[:email_yaml]
   default :from => Proc.new { Setting[:email_reply_address] || "noreply@foreman.example.org" }
+  self.delivery_method = Setting[:delivery_method] unless Setting[:email_yaml]
```

Comment:
These checks on the email_yaml setting are being made at the class level, so will it require a restart to take effect?  Can they be dynamic?